### PR TITLE
feat(tests): create test and refactor code

### PR DESCRIPTION
**Pull request by PiotrFLEURY**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -47,6 +49,7 @@ class MyApp extends StatelessWidget {
   // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
+    GetIt.I.get<PreferencesService>().initSharedPreferences();
```
Comment (zohra051) [inline lib/main.dart]: Please use _prefService = GetIt.I.get<PreferencesService>()

File: android/app/.classpath [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (nathanvandeperre97) [inline android/app/.classpath]: Can't we add this file in the .gitignore ?

**Review by nathanvandeperre97 — CHANGES_REQUESTED** [review]
We need to add the file android/app/.classpath to the .gitignore

File: test/widgets/profile/profile_change_pwd_test.dart [hunk 3]
```diff
@@ -54,21 +59,20 @@ void main() {
...
 
     when(user.photoUrl).thenReturn("");
 
-    Profile widget = Profile();
-    var testableWidget = buildTestableWidget(widget);
-    await tester.pumpWidget(testableWidget);
+    //WHEN
+    await tester.pumpWidget(widget);
     await tester.pumpAndSettle(Duration(seconds: 3));
     await tester.tap(find.byKey(Key("Password")));
```
Comment (zohra051) [inline test/widgets/profile/profile_change_pwd_test.dart]: please use key = "Password"

File: test/widgets/profile/profile_change_pwd_test.dart [hunk 4]
```diff
@@ -54,21 +59,20 @@ void main() {
...
 
+    //THEN
     expect(find.text('Changer mon mot de passe'), findsOneWidget);
```
Comment (zohra051) [inline test/widgets/profile/profile_change_pwd_test.dart:76]: please use text = 'Changer mon mot de passe'

File: lib/main.dart (hunk 1 above)
Comment (zohra051) [inline lib/main.dart]: please use _prefService = GetIt.I.get<PreferencesService>()

File: test_driver/app_test.dart [hunk 5]
```diff
@@ -45,13 +44,19 @@ void main() {
       //expect(value, "My Lists");
     });
 
+    test('bottom bar navigation', () async {
+      await driver.tap(find.byValueKey("Cards"));
```
Comment (zohra051) [inline test_driver/app_test.dart]: please use keyCard = "Cards"

File: test_driver/app_test.dart [hunk 6]
```diff
@@ -45,13 +44,19 @@ void main() {
...
+      await driver.tap(find.byValueKey("Lists"));
```
Comment (zohra051) [inline test_driver/app_test.dart]: please use keyLists = "Lists"

File: test_driver/app_test.dart [hunk 7]
```diff
@@ -45,13 +44,19 @@ void main() {
...
+      await driver.tap(find.byValueKey("Share"));
```
Comment (zohra051) [inline test_driver/app_test.dart]: please use keyShare = "Share"

File: test_driver/app_test.dart [hunk 8]
```diff
@@ -45,13 +44,19 @@ void main() {
...
+      await driver.tap(find.byValueKey("Profile"));
```
Comment (zohra051) [inline test_driver/app_test.dart]: please use keyProfile = "Profile"

File: test_driver/app_test.dart [hunk 9]
```diff
@@ -45,13 +44,19 @@ void main() {
...
+    });
+
     test('Delete this account', () async {
       await driver.tap(find.byValueKey("Profile"));
```
Comment (zohra051) [inline test_driver/app_test.dart]: please use keyProfile = "Profile"

File: test_driver/app_test.dart [hunk 10]
```diff
@@ -45,13 +44,19 @@ void main() {
...
       await driver.tap(find.byValueKey("debug_delete_account_button"));
```
Comment (zohra051) [inline test_driver/app_test.dart]: please use debug_delete_account_button = "debug_delete_account_button"

File: test_driver/app_test.dart [hunk 11]
```diff
@@ -45,13 +44,19 @@ void main() {
...
       String value =
           await driver.getText(find.byValueKey("authentication_page_text"));
```
Comment (zohra051) [inline test_driver/app_test.dart]: please use authentication_page_text = "authentication_page_text"

File: test_driver/app_test.dart [hunk 12]
```diff
@@ -45,13 +44,19 @@ void main() {
...
 
-      await Future<Null>.delayed(Duration(seconds: 2));
       expect(value, "Login to your account");
```
Comment (zohra051) [inline test_driver/app_test.dart:58]: please use login = "Login to your account"

**Review by zohra051 — COMMENTED** [review]
Please correct this files

**Review by nathanvandeperre97 — DISMISSED** [review]

**Review by zohra051 — COMMENTED** [review]
perfect

**Review by nathanvandeperre97 — APPROVED** [review]

**Review by zohra051 — COMMENTED** [review]
ok

**Review by zohra051 — APPROVED** [review]

**Pull request merged**